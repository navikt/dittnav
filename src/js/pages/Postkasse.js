import React, { Component } from 'react';
import DittLink from 'js/components/common/DittLink';
import Tjenester from 'js/components/Tjenester';

class Postkasse extends Component {
  render() {
    const { info } = this.props;
    return (
      <div>
        <h1 className="typo-sidetittel text-center">Din dialog med NAV</h1>
        <Tjenester services={info.viktigeTjenester} />
        <section className="text-center">
          <DittLink url="/" className="lenke">Til Ditt NAV</DittLink>
        </section>
      </div>);
  }
}

Postkasse.propTypes = Tjenester.propTypes;

export default Postkasse;
