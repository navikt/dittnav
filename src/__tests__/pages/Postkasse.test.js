import * as React from 'react';
import Postkasse from 'js/pages/Postkasse';
import wrapIntl from 'js/IntlTestHelper';
const ReactTestRenderer = require('react-test-renderer');

test('Postkasse renders', () => {
  const info = {
    viktigeTjenester: [
      { url: 'https://nav.no/', tittel: 'Din Pensjon', beskrivelse: 'Short desc', bildenavn: 'din_pensjon' },
      { url: 'https://nav.no/abc', tittel: 'Meldekort', beskrivelse: 'Mock desc', bildenavn: 'registrer_arbeidssoker' },
    ],
  };
  const component = ReactTestRenderer.create(wrapIntl(<Postkasse info={info} />));
  expect(component.toJSON()).toMatchSnapshot();
});
