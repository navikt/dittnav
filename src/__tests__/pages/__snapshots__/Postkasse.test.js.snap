// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Postkasse renders 1`] = `
<div>
  <h1
    className="typo-sidetittel text-center"
  >
    Din dialog med NAV
  </h1>
  <section
    className="ditt-list"
  >
    <a
      className="lenke tjeneste-boks ditt-list-element ditt-nav-din-pensjon"
      data-ga="Dittnav/Lenkeboks"
      href="https://nav.no/"
    >
      <h2
        className="underline"
      >
        Din Pensjon
      </h2>
      <p>
        Short desc
      </p>
    </a>
    <a
      className="lenke tjeneste-boks ditt-list-element ditt-nav-registrer-arbeidssoker"
      data-ga="Dittnav/Lenkeboks"
      href="https://nav.no/abc"
    >
      <h2
        className="underline"
      >
        Meldekort
      </h2>
      <p>
        Mock desc
      </p>
    </a>
  </section>
  <section
    className="text-center"
  >
    <a
      className="lenke"
      onClick={[Function]}
    >
      Til Ditt NAV
    </a>
  </section>
</div>
`;
